source/security/authentication.txt

```diff
@@ -0,0 +1,401 @@
+- ``authMechanism``: Set to ``'MONGODB-AWS'``.
+
+You can set these options in two ways: by passing an options array to the
+``MongoDB\Client`` constructor or through parameters in your connection URI.
+
+.. tabs::
+
+   .. tab:: MongoDB\\Client
+      :tabid: Client
+
+      .. code-block:: php
+
+         $options = [
+             'username' => '<AWS IAM access key ID>',
+             'password' => '<AWS IAM secret access key>',
+             'authMechanism' => 'MONGODB-AWS',
+         ];
+
+         $client = new Client(
+            'mongodb://<hostname>:<port>',
+            $options,
+         );
+
+   .. tab:: Connection URI
+      :tabid: connectionstring
+
+      .. code-block:: php
+
+         $uri = 'mongodb://<AWS IAM access key ID>:<AWS IAM secret access key>@<hostname>:<port>/?authMechanism=MONGODB-AWS';
+         $client = new Client($uri);
+
+.. _php-mongodb-aws-env-vars:
+
+Environment Variables
+~~~~~~~~~~~~~~~~~~~~~
+
+If you don't provide a username and password when you construct your ``MongoDB\Client``
+object, the {+php-library+} tries to retrieve AWS credentials from the following
+environment variables:
+
+- ``AWS_ACCESS_KEY_ID``
+- ``AWS_SECRET_ACCESS_KEY``
+- ``AWS_SESSION_TOKEN``
+
+To use these environment variables to authenticate your application, first set them to the
+AWS IAM values needed for authentication, as shown in the following code
+example:
+
+.. code-block:: sh
+
+   export AWS_ACCESS_KEY_ID=<AWS IAM access key ID>
+   export AWS_SECRET_ACCESS_KEY=<AWS IAM secret access key>
+   export AWS_SESSION_TOKEN=<AWS session token>
+
+.. important::
+  
+   Don't percent-encode the values in these environment variables.
+
+After you set these environment variables, set the ``authMechanism``
+connection option to ``'MONGODB-AWS'``.
+
+.. _php-mongodb-aws-env-example:
+
+Example
+```````
+
+The following example sets the ``authMechanism`` connection option. 
+You can set this option in two ways: by passing an options array to the
+``MongoDB\Client`` constructor or through a parameter in your connection URI.
+
+.. tabs::
+
+   .. tab:: MongoDB\\Client
+      :tabid: Client
+
+      .. code-block:: php
+
+         $client = new Client(
+            'mongodb://<hostname>:<port>',
+            ['authMechanism' => 'MONGODB-AWS'],
+         );
+
+   .. tab:: Connection URI
+      :tabid: connectionstring
+
+      .. code-block:: php
+
+         $uri = 'mongodb://<hostname>:<port>/?&authMechanism=MONGODB-AWS';
+         $client = new Client($uri);
+
+.. tip:: AWS Lambda
+   
+   AWS Lambda runtimes can automatically set these environment variables during
+   initialization. For more information about using environment variables in an AWS Lambda
+   environment, see `Using Lambda environment variables
+   <https://docs.aws.amazon.com/lambda/latest/dg/configuration-envvars.html#configuration-envvars-runtime>`__
+   in the AWS documentation.
+
+.. _php-mongodb-aws-oidc:
+
+AssumeRoleWithWebIdentity Request
```

Comment:
Note: PHP doesn't test any of this, as we rely entirely on the C driver's implementation. See [PHPC-1895](https://jira.mongodb.org/browse/PHPC-1895) for more context.

I don't consider myself qualified to review this, but if the text applies to the C driver docs it should certainly work for us. And I assume that's the case since there's no PHP code here.

In all of these AWS examples, the _only_ thing handled by PHP is constructing the client object from a connection string and any URI options. The C driver handles all reading of AWS environment variables and any API requests like `AssumeRoleWithWebIdentity`.

Reply:
I added a note about this in the AWS intro section. Is it accurate to say that the C driver retrieves AWS credentials from each source in this section?

Reply:
When the Manager and Client constructors receive their connection string (`$uri`) and extra options (`$uriOptions`), we pass the connection string directly to libmongoc for parsing and then iteratively apply each field in `$uriOptions` to the parsed URI object. That object is then used to create the libmongoc client.

For that reason, I wouldn't suggest any language about the "C driver" doing anything with a Client constructor. The intro section note about PHP relying on libmongoc for AWS auth mechanisms seems sufficient, and then we can just refer to the PHP driver in subsequent sections. The link to the C driver docs for AWS auth is helpful, so thanks for adding that.